cmake: thrift.cmake: should not include the system header path

Compile error
```
[50/106] Building CXX object CMakeFiles/app.dir/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.cpp.obj
FAILED: CMakeFiles/app.dir/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.cpp.obj 
ccache /Users/cfriedt/zephyr-sdk-0.13.1/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-g++ -DKERNEL -D_FORTIFY_SOURCE=2 -D_GLIBCXX_HAS_GTHREADS=1 -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/Users/cfriedt/workspace/thrift-for-zephyr-workspace/thrift-for-zephyr/build/zephyr/misc/generated/thrift_hello/gen-cpp -I/Users/cfriedt/homebrew/Cellar/thrift/0.16.0/include -I/Users/cfriedt/workspace/thrift-for-zephyr-workspace/zephyr/include -I/Users/cfriedt/workspace/thrift-for-zephyr-workspace/thrift-for-zephyr/build/zephyr/include/generated -I/Users/cfriedt/workspace/thrift-for-zephyr-workspace/zephyr/soc/x86/ia32 -I/Users/cfriedt/workspace/thrift-for-zephyr-workspace/zephyr/lib/libc/newlib/include -I/Users/cfriedt/workspace/thrift-for-zephyr-workspace/zephyr/subsys/testsuite/include -I/Users/cfriedt/workspace/thrift-for-zephyr-workspace/zephyr/subsys/testsuite/ztest/include -I/Users/cfriedt/workspace/thrift-for-zephyr/lib/thrift/include -m64  -Os -fcheck-new -std=c++17 -Wno-register -fno-exceptions -fno-rtti -imacros /Users/cfriedt/workspace/thrift-for-zephyr-workspace/thrift-for-zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -m64 -Wa,--divide -imacros /Users/cfriedt/workspace/thrift-for-zephyr-workspace/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/Users/cfriedt/workspace/thrift-for-zephyr-workspace/thrift-for-zephyr/tests/lib/thrift/hello=CMAKE_SOURCE_DIR -fmacro-prefix-map=/Users/cfriedt/workspace/thrift-for-zephyr-workspace/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/Users/cfriedt/workspace/thrift-for-zephyr-workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -mno-sse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -mno-sse4a -MD -MT CMakeFiles/app.dir/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.cpp.obj -MF CMakeFiles/app.dir/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.cpp.obj.d -o CMakeFiles/app.dir/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.cpp.obj -c /Users/cfriedt/workspace/thrift-for-zephyr-workspace/thrift-for-zephyr/build/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.cpp
In file included from /Users/cfriedt/homebrew/Cellar/thrift/0.16.0/include/thrift/transport/TTransport.h:23,
                 from /Users/cfriedt/homebrew/Cellar/thrift/0.16.0/include/thrift/protocol/TProtocol.h:28,
                 from /Users/cfriedt/homebrew/Cellar/thrift/0.16.0/include/thrift/TProcessor.h:24,
                 from /Users/cfriedt/homebrew/Cellar/thrift/0.16.0/include/thrift/TDispatchProcessor.h:22,
                 from /Users/cfriedt/workspace/thrift-for-zephyr-workspace/thrift-for-zephyr/build/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.h:10,
                 from /Users/cfriedt/workspace/thrift-for-zephyr-workspace/thrift-for-zephyr/build/zephyr/misc/generated/thrift_hello/gen-cpp/Hello.cpp:7:
/Users/cfriedt/homebrew/Cellar/thrift/0.16.0/include/thrift/Thrift.h:32:10: fatal error: netinet/in.h: No such file or directory
   32 | #include <netinet/in.h>
      |          ^~~~~~~~~~~~~~

```

It looks as though `#include <thrift/Thrift.h>` is resolved to the system thrift include path with `cmake/thrift.cmake`
```
pkg_search_module(THRIFT REQUIRED thrift)
```

This is something we will want to fix after the submodule is introduced.

--- a/cmake/thrift.cmake
+++ b/cmake/thrift.cmake
@@ -1,8 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 find_program(THRIFT_EXECUTABLE thrift)
-find_package(PkgConfig REQUIRED)
-pkg_search_module(THRIFT REQUIRED thrift)
 
 message(STATUS "Found thrift executable: ${THRIFT_EXECUTABLE}")
 message(STATUS "Found pkg-config executable: ${PKGCONFIG_EXECUTABLE}")
@@ -30,5 +28,4 @@ function(thrift
     )
 
     target_include_directories(${target} PRIVATE ${out_dir}/gen-${lang})
-    target_include_directories(${target} PRIVATE ${THRIFT_INCLUDE_DIRS})
 endfunction()

